compiler/src/iree/compiler/Codegen/SPIRV/AMDConfig.cpp

```diff
@@ -82,8 +82,12 @@ LogicalResult setAMDCodeGenConfig(const spirv::TargetEnv &targetEnv,
   }
 
   if (auto convOp = dyn_cast<linalg::ConvolutionOpInterface>(rootOp)) {
+    auto type = convOp.image().getType().cast<ShapedType>();
```

Comment:
nit: Prefer free cast functions (https://mlir.llvm.org/deprecation/)

Also in the other config code below